modules/admission-webhook-create.adoc

```diff
@@ -0,0 +1,328 @@
+  kind: ClusterRoleBinding
+  metadata:
+    name: auth-delegator-<namespace_name>
+  roleRef:
+    kind: ClusterRole
+    apiGroup: rbac.authorization.k8s.io
+    name: system:auth-delegator
+  subjects:
+  - kind: ServiceAccount
+    namespace: <namespace_name>
+    name: server
+
+- apiVersion: rbac.authorization.k8s.io/v1  <2>
+  kind: ClusterRole
+  metadata:
+    annotations:
+    name: system:openshift:online:<webhook-name>-server
+  rules:
+  - apiGroups:
+    - online.openshift.io
+    resources:
+    - <resource_name>
+    verbs:
+    - get
+    - list
+    - watch
+
+- apiVersion: rbac.authorization.k8s.io/v1  <3>
+  kind: ClusterRole
+  metadata:
+    name: system:openshift:online:<webhook_name>-requester
+  rules:
+  - apiGroups:
+    - admission.online.openshift.io
+    resources:
+    - <resource_name>
+    verbs:
+    - create
+
+- apiVersion: rbac.authorization.k8s.io/v1  <4>
+  kind: ClusterRoleBinding
+  metadata:
+    name: namespace-reservation-server-<namespace_name>
+  roleRef:
+    kind: ClusterRole
+    apiGroup: rbac.authorization.k8s.io
+    name: system:openshift:online:<webhook_name>-server
+  subjects:
+  - kind: ServiceAccount
+    namespace: <namespace_name>
+    name: server
+
+- apiVersion: rbac.authorization.k8s.io/v1  <5>
+  kind: RoleBinding
+  metadata:
+    namespace: kube-system
+    name: extension-server-authentication-reader-<namespace_name>
+  roleRef:
+    kind: Role
+    apiGroup: rbac.authorization.k8s.io
+    name: extension-apiserver-authentication-reader
+  subjects:
+  - kind: ServiceAccount
+    namespace: <namespace_name>
+    name: server
+
+- apiVersion: rbac.authorization.k8s.io/v1  <6>
+  kind: ClusterRole
+  metadata:
+    name: <aggregated_apiserver_cluster_role>
+  rules:
+  - apiGroups:
+    - admissionregistration.k8s.io
+    resources:
+    - validatingwebhookconfigurations
+    - mutatingwebhookconfigurations
+    verbs:
+    - get
+    - list
+    - watch
+  - apiGroups:
+    - ""
+    resources:
+    - namespaces
+    verbs:
+    - get
+    - list
+    - watch
+
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRoleBinding
+  metadata:
+    name: <aggregated_apiserver_cluster_role>
+  roleRef:
+    kind: ClusterRole
+    apiGroup: rbac.authorization.k8s.io
+    name: <aggregated_apiserver_cluster_role>
+  subjects:
+  - kind: ServiceAccount
+    namespace: <namespace_name>
+    name: server
+----
+<1> Delegates authentication and authorization to the webhook server API.
+<2> Allows the admission webhook server to access cluster resource.
+<3> Enables the aggregated API server to create admission reviews.
+<4> Also enables the admission server to access cluster resource.
+<5> Role binding to read the configuration for terminating authentication.
+<6> Default cluster role and cluster role bindings for an aggregated API server.
+
+. Apply those RBAC rules to the cluster:
++
+----
+$ oc auth reconcile -f <YAML_filename>
+----
+
+. Create a new project to run the DaemonSet:
```

Comment:
Is this namespace the same namespace that is used in the YAML above? If so, wouldn't it need to have been created already before that YAML file is applied?

Reply:
Also, this says to create a project, but then technically creates a namespace. I would say namespace here instead of project.

Reply:
Thank you, I will use 'namespace' from now on and have replaced references to 'project' in the procedure. These will be pushed to the PR shortly, once I have made a few other changes.

Regarding creating the namespace after defining the RBAC resources, the document follows the same order as the script in https://github.com/pneedle-rh/kubernetes-namespace-reservation/blob/master/hack/install-kube.sh, which I tested successfully previously. I will run through another test sequence once I have finished making each of the recommended changes, to ensure that functionality is as expected.

Reply:
On reflection, I think that even though the sequence works in testing, it reads better anyway if the namespace is created as one of the first steps. I have moved that step up in the sequence.